File: backend/dto/wallet.go
package dto

import "github.com/shopspring/decimal"

type WalletDTO struct {
	Id                        int              `json:"id"`
	Name                      string           `json:"name"`
	Crypto                    string           `json:"crypto"`
	Fiat                      string           `json:"fiat"`
	CryptoUnitValue           decimal.Decimal  `json:"cryptoUnitValue"`
	TotalCryptoAmount         decimal.Decimal  `json:"totalCryptoAmount"`
	TotalFiatInvested         decimal.Decimal  `json:"totalFiatInvested"`
	CurrentFiatValue          decimal.Decimal  `json:"currentFiatValue"`
	ReturnOnInvestment        decimal.Decimal  `json:"returnOnInvestment"`
	ReturnOnInvestmentPercent decimal.Decimal  `json:"returnOnInvestmentPercent"`
	CryptoUnitValueBreakEven  decimal.Decimal  `json:"cryptoUnitValueBreakEven"`
	Transactions              []TransactionDTO `json:"transactions"`
}

File: frontend/src/app/client/model/wallet.ts
export interface Wallet {
    id: number
    name: string
    crypto: string
    fiat: string
    cryptoUnitValue: string
    totalCryptoAmount: string
    totalFiatInvested: string
    currentFiatValue: string
    returnOnInvestment: string
    returnOnInvestmentPercent: string
    cryptoUnitValueBreakEven: string
    transactions: Transaction[]
}

export interface Transaction {
    id: number
    walletId: number
    time: Date
    cryptoValue: string
    cryptoAmount: string
    fiatInvested: string
    fiatValue?: string
    totalCryptoAmount?: string
    totalFiatInvested?: string
    fiatWalletValue?: string
    fiatReturnOnInvestment?: string
}